src/nvim/search.c

```diff
@@ -3767,7 +3764,7 @@ void find_pattern_in_path(char_u *ptr, Direction dir, size_t len, bool whole, bo
 
         if (compl_status_adding() && i == ins_compl_len()) {
           // IOSIZE > compl_length, so the STRNCPY works
-          STRNCPY(IObuff, aux, i);
```

Comment:
Better add `NOLINT`

Reply:
How come? Do we want `STRNCPY` here?

Reply:
It is from Vim isn't it?

Reply:
Yes, why does that matter?

Reply:
Look at #6008. It seems that this has already been discussed.

Reply:
> several call sites are depending on the zero-padding behavior of strncpy.

I'd leave the remaining strncpy calls alone.